db-engine/src/main/java/com/splicemachine/db/catalog/types/TypeDescriptorImpl.java

```diff
@@ -65,6 +65,8 @@
     private int                        scale;
     private boolean                    isNullable;
     private int                        maximumWidth;
+
+    transient private boolean          forSbcsData;
```

Comment:
@arnaud-splice have you tested this with triggers? 
@jyuanca @yxia92 do you think this could cause issues if we don't serialize it?

Reply:
I have, everything works well. We only use this variable in `CastNode.generateExpression`. I think it'd be cleaner to add this boolean to the cast node itself, but that means reworking the grammar quite a bit, I'll see what I can do there.

Reply:
Refactoring done.